File: backbone/test/test_service.py
from service import Service

def on_request(msg, reply):
    try:
        reply(['test', 'reply'])
    except Exception as e:
        print "Caught Exception", type(e), e
        reply("Exception")
    except TypeError as e:
        print "Caught Error", str(e)
        reply("ERROR")


if __name__ == '__main__':
    import baker
    @baker.command(default=True)
    def test():
        s = Service('test', on_request)
        s.start()

    baker.run()

File: backbone/test/test_service.js
var BackboneService = require('dsp_service');

if (require.main === module) {
  var baker = require('dsp_shared/lib/baker');  
  baker.command(function test() {
    console.log("TESTING");
    var backbone = new BackboneService('test', function(message, reply){
      console.log("MESSAGE", message);
      reply(['test-test', 'reply']);
    });
    backbone.connect();
    
  }, {default: true});
  baker.run();
}
